lib/DatabaseLayer.py

```diff
@@ -176,7 +178,8 @@ def cvesForCPE(
                 cpe_version = re_from_start.sub("", vc)
 
                 # TODO: handle versions such as "1.1.3:p2"
-                cpe_version = cpe_version.split(":")[0]
+                # Should work now, I don't know the context of this code.
+                cpe_version = split_cpe_name(cpe_version)[0]
```

Comment:
Revise the context around this. Right after this the `cpe_version` gets replaced by the *first match* of regex `([0-9\.]*` in it, which will remove the `:` and anything after it, making this modification useless. If you can come up with a solution that covers both the `TODO`'s , feel free to write it and remove those comments.

Reply:
looking into it

Reply:
Sadly, I don't think I will be fixing this. After looking into it, the fix seems kinda complicated and would only affect a single flag of `bin/search.py`, which I don't use.

Reply:
Thanks for the try! I don't think this PR makes the situation any worse than it is, so I would accept it as is and leave the TODOs for someone else.